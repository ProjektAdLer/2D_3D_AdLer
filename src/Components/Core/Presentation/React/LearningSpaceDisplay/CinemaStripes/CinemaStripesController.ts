import ICinemaStripesController from "./ICinemaStripesController";
export default class CinemaStripesController
  implements ICinemaStripesController {}
