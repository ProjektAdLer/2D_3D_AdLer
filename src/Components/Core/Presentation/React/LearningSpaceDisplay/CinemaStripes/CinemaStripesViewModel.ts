import Observable from "src/Lib/Observable";

export default class CinemaStripesViewModel {
  isOpen: Observable<boolean> = new Observable<boolean>(false);
}
