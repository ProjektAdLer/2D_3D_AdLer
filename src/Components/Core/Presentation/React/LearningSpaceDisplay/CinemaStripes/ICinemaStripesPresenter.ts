import { StoryElementType } from "src/Components/Core/Domain/Types/StoryElementType";

export default interface ICinemaStripesPresenter {
  onStoryElementCutSceneTriggered(storyType: StoryElementType): void;
  onStoryElementCutSceneFinished(): void;
}
