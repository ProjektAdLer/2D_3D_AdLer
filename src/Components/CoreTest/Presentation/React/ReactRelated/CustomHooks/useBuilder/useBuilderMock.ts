import * as useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";

/**
 * Mocks the useViewModelControllerProvider hook.
 * @param mockedTupels The return value, which the mocked hook will return.
 *
 */
export default function useBuilderMock(
  mockedTupels: [viewModel: any, controller: any] = [undefined, undefined]
) {
  jest.spyOn(useBuilder, "default").mockReturnValue(mockedTupels);
}
