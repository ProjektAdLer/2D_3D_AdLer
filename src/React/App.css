body {
  margin: 0;
  height: 100vh;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

div{
  width: 100%;
  height: 100%;
}

section{
  display: inline;
  position: fixed;
  top: 0;
  left: 0;
  width: 120px;
  height: 100vh;
  background-color: #fff;
}

button{
  width: 100px;
  margin: 10px;
  z-index: 5;
  position: fixed;
  left: 0;
  top: 0;
}

canvas {
  width: calc(100% - 100px);
  height: 100%;
  margin-left: 100px;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.modal {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
}

.modal-content {
  width: 1000px;
  background-color: #fff;
}

.modal-header,
.modal-footer {
  padding: 10px;
  height: 60px;
}

.modal-title {
  margin: 0;
}

.modal-body {
  padding: 10px;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
}
