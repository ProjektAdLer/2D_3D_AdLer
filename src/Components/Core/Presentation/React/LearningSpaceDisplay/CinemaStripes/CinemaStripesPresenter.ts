import { StoryElementType } from "src/Components/Core/Domain/Types/StoryElementType";
import ICinemaStripesPresenter from "./ICinemaStripesPresenter";
import CinemaStripesViewModel from "./CinemaStripesViewModel";
export default class CinemaStripesPresenter implements ICinemaStripesPresenter {
  constructor(private viewModel: CinemaStripesViewModel) {}

  onStoryElementCutSceneTriggered(storyType: StoryElementType): void {
    this.viewModel.isOpen.Value = true;
  }
  onStoryElementCutSceneFinished(): void {
    this.viewModel.isOpen.Value = false;
  }
}
