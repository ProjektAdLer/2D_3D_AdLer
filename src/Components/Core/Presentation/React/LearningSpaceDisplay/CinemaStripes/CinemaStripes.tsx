import useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";
import CinemaStripesViewModel from "./CinemaStripesViewModel";
import ICinemaStripesController from "./ICinemaStripesController";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";
import useObservable from "~ReactComponents/ReactRelated/CustomHooks/useObservable";

export default function CinemaStripes() {
  const [viewModel, controller] = useBuilder<
    CinemaStripesViewModel,
    ICinemaStripesController
  >(BUILDER_TYPES.ICinemaStripesBuilder);

  const [isOpen] = useObservable<boolean>(viewModel?.isOpen);

  if (!viewModel || !controller) return null;
  if (!isOpen) return null;
  return (
    <div>
      <div className="absolute top-0 left-0 w-full h-24 bg-black z-[100]"></div>
      <div className="absolute bottom-0 left-0 w-full h-24 bg-black z-[100]"></div>
    </div>
  );
}
