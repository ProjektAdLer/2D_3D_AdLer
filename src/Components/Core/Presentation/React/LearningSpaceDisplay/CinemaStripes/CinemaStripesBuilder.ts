import { injectable } from "inversify";
import PresentationBuilder from "../../../PresentationBuilder/PresentationBuilder";
import CinemaStripesViewModel from "./CinemaStripesViewModel";
import CinemaStripesController from "./CinemaStripesController";
import CinemaStripesPresenter from "./CinemaStripesPresenter";
import ICinemaStripesController from "./ICinemaStripesController";
import ICinemaStripesPresenter from "./ICinemaStripesPresenter";
import CoreDIContainer from "~DependencyInjection/CoreDIContainer";
import PRESENTATION_TYPES from "~DependencyInjection/Presentation/PRESENTATION_TYPES";
import ILearningWorldPort from "src/Components/Core/Application/Ports/Interfaces/ILearningWorldPort";
import PORT_TYPES from "~DependencyInjection/Ports/PORT_TYPES";
import { LocationScope } from "~ReactComponents/ReactRelated/ReactEntryPoint/HistoryWrapper";

@injectable()
export default class CinemaStripesBuilder extends PresentationBuilder<
  CinemaStripesViewModel,
  ICinemaStripesController,
  undefined,
  ICinemaStripesPresenter
> {
  constructor() {
    super(
      CinemaStripesViewModel,
      CinemaStripesController,
      undefined,
      CinemaStripesPresenter,
    );
  }
  override buildPresenter(): void {
    super.buildPresenter();
    CoreDIContainer.bind<ICinemaStripesPresenter>(
      PRESENTATION_TYPES.ICinemaStripesPresenter,
    ).toConstantValue(this.presenter!);

    CoreDIContainer.get<ILearningWorldPort>(
      PORT_TYPES.ILearningWorldPort,
    ).registerAdapter(this.presenter!, LocationScope._sceneRendering);
  }
  override buildViewModel(): void {
    super.buildViewModel();
  }
  override buildController(): void {
    super.buildController();
  }
}
