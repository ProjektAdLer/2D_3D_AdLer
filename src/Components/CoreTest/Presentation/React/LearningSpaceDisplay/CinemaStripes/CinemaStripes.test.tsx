import React from "react";
import useBuilderMock from "../../ReactRelated/CustomHooks/useBuilder/useBuilderMock";
import { render } from "@testing-library/react";
import CinemaStripes from "../../../../../Core/Presentation/React/LearningSpaceDisplay/CinemaStripes/CinemaStripes";
import CinemaStripesViewModel from "../../../../../Core/Presentation/React/LearningSpaceDisplay/CinemaStripes/CinemaStripesViewModel";
import CinemaStripesController from "../../../../../Core/Presentation/React/LearningSpaceDisplay/CinemaStripes/CinemaStripesController";
import { mock } from "jest-mock-extended";

let viewModel = new CinemaStripesViewModel();
let fakeController = mock<CinemaStripesController>();
describe("CinemaStripes", () => {
  test("doesn't render without controller", () => {
    useBuilderMock([viewModel, undefined]);
    const { container } = render(<CinemaStripes />);
    expect(container.firstChild).toBeNull();
  });

  test("doesn't render without view model", () => {
    useBuilderMock([undefined, fakeController]);
    const { container } = render(<CinemaStripes />);
    expect(container.firstChild).toBeNull();
  });

  test("doesn't render if isOpen is set to false", () => {
    viewModel.isOpen.Value = false;
    useBuilderMock([viewModel, fakeController]);
    const { container } = render(<CinemaStripes />);
    expect(container.firstChild).toBeNull();
  });

  test("renders if isOpen is set to true", () => {
    viewModel.isOpen.Value = true;
    useBuilderMock([viewModel, fakeController]);
    const { container } = render(<CinemaStripes />);
    expect(container.childElementCount).toBe(1);
    expect(container).toMatchSnapshot();
  });
});
