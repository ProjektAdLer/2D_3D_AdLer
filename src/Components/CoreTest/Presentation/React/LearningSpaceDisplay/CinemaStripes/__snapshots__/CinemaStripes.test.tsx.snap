// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CinemaStripes renders if isOpen is set to true 1`] = `
<div>
  <div>
    <div
      class="absolute top-0 left-0 w-full h-24 bg-black z-[100]"
    />
    <div
      class="absolute bottom-0 left-0 w-full h-24 bg-black z-[100]"
    />
  </div>
</div>
`;
